medusa/search/core.py

```diff
@@ -805,41 +805,20 @@ def collect_multi_candidates(candidates, series_obj, episodes, down_cur_quality)
     if not wanted_candidates:
         return multi_candidates, single_candidates
 
-    searched_seasons = {str(x.season) for x in episodes}
-    main_db_con = db.DBConnection()
-    selection = main_db_con.select(
-        'SELECT episode '
-        'FROM tv_episodes '
-        'WHERE indexer = ?'
-        ' AND showid = ?'
-        ' AND ( season IN ( {0} ) )'.format(','.join(searched_seasons)),
-        [series_obj.indexer, series_obj.series_id]
-    )
-    all_eps = [int(x[b'episode']) for x in selection]
-    log.debug(u'Episodes list: {0}', all_eps)
-
     for candidate in wanted_candidates:
-        season_quality = candidate.quality
-
         all_wanted = True
         any_wanted = False
-        for cur_ep_num in all_eps:
-            for season in {x.season for x in episodes}:
-                if not series_obj.want_episode(season, cur_ep_num, season_quality,
-                                               down_cur_quality):
-                    all_wanted = False
-                else:
-                    any_wanted = True
+        for ep_obj in candidate.episodes:
```

Comment:
How do you feel about this?
```py
wanted_check = (
    series_obj.want_episode(ep_obj.season, ep_obj.episode, candidate.quality, down_cur_quality)
    for ep_obj in candidate.episodes
)
all_wanted = all(wanted_check)
any_wanted = any(wanted_check)
```

And since `all_wanted`/`any_wanted` are only used once,
lose the variables and just `if all(wanted_check): ... elif any(wanted_check):`
It would also make the checks lazy.

**I didn't test my code.**